### PR TITLE
feat: add cas apis

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: packages/core/src/internal/clients/cache/AbstractCacheClient.ts [hunk 1]
```diff
@@ -682,6 +694,180 @@ export abstract class AbstractCacheClient implements ICacheClient {
...
+   * @param {string} cacheName - The cache to store the value in.
+   * @param {string | Uint8Array} key - The key to set.
+   * @param {string | Uint8Array} field - The value to be stored.
+   * @param {SetIfAbsentOptions} [options]
+   * @param {number} [options.ttl] - The time to live for the item in the cache.
+   * Uses the client's default TTL if this is not supplied.
+   * @returns {Promise<CacheSetIfAbsent.Response>} -
+   * {@link CacheSetIfAbsent.Stored} on storing the new value.
+   * {@link CacheSetIfAbsent.NotStored} on not storing the new value.
+   * {@link CacheSetIfAbsent.Error} on failure.
```
Comment (anitarua) [inline packages/core/src/internal/clients/cache/AbstractCacheClient.ts:733]: SetIfAbsent -> SetIfPresent

File: packages/core/src/internal/clients/cache/AbstractCacheClient.ts [hunk 2]
```diff
@@ -682,6 +694,180 @@ export abstract class AbstractCacheClient implements ICacheClient {
...
+   * if the value for the key is not equal to the supplied `notEqual` parameter.
+   *
+   * @param {string} cacheName - The cache to store the value in.
+   * @param {string | Uint8Array} key - The key to set.
+   * @param {string | Uint8Array} field - The value to be stored.
+   * @param {string | Uint8Array} notEqual - The value to compare to the cached value.
+   * @param {SetIfNotEqualOptions} [options]
+   * @param {number} [options.ttl] - The time to live for the item in the cache.
+   * Uses the client's default TTL if this is not supplied.
+   * @returns {Promise<CacheSetIfAbsent.Response>} -
```
Comment (anitarua) [inline packages/core/src/internal/clients/cache/AbstractCacheClient.ts:783]:
```suggestion
   * @returns {Promise<CacheSetIfNotEqual.Response>} -
```

File: packages/core/src/internal/clients/cache/AbstractCacheClient.ts [hunk 3]
```diff
@@ -682,6 +694,180 @@ export abstract class AbstractCacheClient implements ICacheClient {
...
+   * @param {string | Uint8Array} key - The key to set.
+   * @param {string | Uint8Array} field - The value to be stored.
+   * @param {string | Uint8Array} notEqual - The value to compare to the cached value.
+   * @param {SetIfAbsentOptions} [options]
+   * @param {number} [options.ttl] - The time to live for the item in the cache.
+   * Uses the client's default TTL if this is not supplied.
+   * @returns {Promise<CacheSetIfAbsent.Response>} -
+   * {@link CacheSetIfAbsent.Stored} on storing the new value.
+   * {@link CacheSetIfAbsent.NotStored} on not storing the new value.
+   * {@link CacheSetIfAbsent.Error} on failure.
```
Comment (anitarua) [inline packages/core/src/internal/clients/cache/AbstractCacheClient.ts:819]: more SetIfAbsent copypasta

File: packages/client-sdk-web/src/internal/cache-data-client.ts [hunk 4]
```diff
@@ -340,6 +355,8 @@ export class CacheDataClient<
     });
   }
 
+  // setIfNotExists is deprecated on the service. Here we call the new `SetIf` method with the absent field set
```
Comment (anitarua) [inline packages/client-sdk-web/src/internal/cache-data-client.ts:358]: I don't see the corresponding change for calling SetIfAbsent in sendSetIfNotExists?

**Review by anitarua — APPROVED** [review]
lgtm!

File: packages/core/src/clients/ICacheClient.ts [hunk 5]
```diff
@@ -109,6 +121,46 @@ export interface ICacheClient extends IControlClient, IPingClient {
     field: string | Uint8Array,
     options?: SetIfNotExistsOptions
   ): Promise<CacheSetIfNotExists.Response>;
+  setIfAbsent(
```
Comment (cprice404) [inline packages/core/src/clients/ICacheClient.ts:124]: can we add a `deprecated` annotation on the `setIfNotExists`?

Comment (cprice404) [reply packages/core/src/clients/ICacheClient.ts:124]: might need to appear in a few different files.

File: packages/core/src/internal/clients/cache/AbstractCacheClient.ts [hunk 6]
```diff
@@ -658,9 +670,9 @@ export abstract class AbstractCacheClient implements ICacheClient {
   }
 
   /**
-   * Associates the given key with the given value. If a value for the key is
-   * already present it is not replaced with the new value.
+   * Associates the given key with the given value if key is not already present in the cache.
    *
+   * @deprecated Use setIfAbsent instead.
```
Comment (cprice404) [inline packages/core/src/internal/clients/cache/AbstractCacheClient.ts:675]: oh I see, you did it here, thanks. Probably a good idea to do it in the interface too.

**Review by anitarua — APPROVED** [review]

**Review by cprice404 — APPROVED** [review]
yay

**Pull request merged**